DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/query9/query9_kernel.cpp

```diff
@@ -383,18 +385,17 @@ bool SubmitQuery9(queue& q, Database& dbinfo, std::string colour,
 
     h.single_task<Compute>([=]() [[intel::kernel_args_restrict]] {
       // the accumulators
-      constexpr int kAccumCacheSize = 7;
-      BRAMAccumulator<DBDecimal, (25 * 7), kAccumCacheSize, unsigned char>
-          sum_profit_local[kFinalDataMaxSize];
+      constexpr int kAccumCacheSize = 8;
+      NTuple<kFinalDataMaxSize, fpga_tools::OnchipMemoryWithCache<
+                                    DBDecimal, (25 * 7), kAccumCacheSize>>
+          sum_profit_local;
 
       // initialize the accumulators
       UnrolledLoop<0, kFinalDataMaxSize>([&](auto j) {
-        sum_profit_local[j].Init();
+        sum_profit_local.template get<j>().init(0);
       });
 
       bool done = false;
-
-      [[intel::initiation_interval(1), intel::ivdep(kAccumCacheSize)]]
```

Comment:
Probably should leave the ii1 attribute here?

Reply:
Same question as in query11

Reply:
same answer :-)